src/System.IO.FileSystem/ref/System.IO.FileSystem.cs

```diff
@@ -212,3 +224,75 @@ public enum SearchOption
+        public string RootDirectory { get { throw null; } }
+        public string OriginalRootDirectory { get { throw null; } }
+        public ReadOnlySpan<char> FileName { get { throw null; } }
+        public FileAttributes Attributes { get { throw null; } }
+        public long Length { get { throw null; } }
+        public DateTimeOffset CreationTimeUtc { get { throw null; } }
+        public DateTimeOffset LastAccessTimeUtc { get { throw null; } }
+        public DateTimeOffset LastWriteTimeUtc { get { throw null; } }
+        public bool IsDirectory { get { throw null; } }
+        public FileSystemInfo ToFileSystemInfo() { throw null; }
+        public string ToSpecifiedFullPath() { throw null; }
+        public string ToFullPath() { throw null; }
+    }
+    public abstract class FileSystemEnumerator<TResult> : Runtime.ConstrainedExecution.CriticalFinalizerObject, Collections.Generic.IEnumerator<TResult>
+    {
+        public FileSystemEnumerator(string directory, EnumerationOptions options = null) { }
+
+        protected virtual bool ShouldIncludeEntry(ref FileSystemEntry entry) { throw null; }
+        protected virtual bool ShouldRecurseIntoEntry(ref FileSystemEntry entry) { throw null; }
+        protected abstract TResult TransformEntry(ref FileSystemEntry entry);
+        protected virtual void OnDirectoryFinished(ReadOnlySpan<char> directory) { throw null; }
+        protected virtual bool ContinueOnError(int error) { throw null; }
+
+        public TResult Current { get { throw null; } }
+        object System.Collections.IEnumerator.Current { get { throw null; } }
+        public bool MoveNext() { throw null; }
+        public void Reset() { throw null; }
+        public void Dispose() { throw null; }
+        protected virtual void Dispose(bool disposing) { throw null; }
+    }
+    public class FileSystemEnumerable<TResult> : Collections.Generic.IEnumerable<TResult>
+    {
+        public FileSystemEnumerable(string directory, FindTransform transform, EnumerationOptions options = null) { }
+
+        public FindPredicate ShouldRecursePredicate { get { throw null; } set { } }
+        public FindPredicate ShouldIncludePredicate { get { throw null; } set { } }
+
+        public Collections.Generic.IEnumerator<TResult> GetEnumerator() { throw null; }
+        Collections.IEnumerator Collections.IEnumerable.GetEnumerator() { throw null; }
+
+
+        public delegate bool FindPredicate(ref FileSystemEntry entry);
+        public delegate TResult FindTransform(ref FileSystemEntry entry);
+    }
+    public static class FileSystemName
+    {
+        public static string TranslateDosExpression(string expression) { throw null; }
+        public static bool MatchesDosExpression(string expression, ReadOnlySpan<char> name, bool ignoreCase = true) { throw null; }
+        public static bool MatchesSimpleExpression(string expression, ReadOnlySpan<char> name, bool ignoreCase = true) { throw null; }
```

Comment:
Why are the input expressions strings rather than spans?

Reply:
Good point. I'll change them, thanks.

Reply:
The Translate one I'm going to keep as string as it doesn't change in most cases. I've optimized to avoid the allocation in that case.